### PR TITLE
Fixed chi2 with thcovmat in report 

### PR DESCRIPTION
**Pull request by andreab1997**
At the moment the thcovmat is not considered for the computation of the chi2 in the report even if the flag use_thcovmat_if_present is True in vp-comparefits.
With this, the full thcovmat is used to compute the total chi2 and an appropriate cut thcovmat is used to compute the chi2 by process or by experiment (or even by dataset).

### CODE REVIEW

**Comment by andreab1997** [comment]
@Zaharid This is the PR against master you requested.

File: validphys2/src/validphys/covmats.py [hunk 1]
```diff
@@ -535,9 +535,15 @@ def pdferr_plus_covmat(dataset, pdf, covmat_t0_considered):
...
-def pdferr_plus_dataset_inputs_covmat(data, pdf, dataset_inputs_covmat_t0_considered):
+def pdferr_plus_dataset_inputs_covmat(data, pdf, dataset_inputs_covmat_t0_considered, fitthcovmat):
     """Like `pdferr_plus_covmat` except for an experiment"""
     # do checks get performed here?
+    if fitthcovmat is not None:
+        theory_covmat = fitthcovmat.load()
+        #change ordering according to exp_covmat (so according to runcard order)
+        tmp = theory_covmat.droplevel(0, axis=0).droplevel(0, axis=1)
+        bb = [str(i) for i in data]
+        return pdferr_plus_covmat(data, pdf, dataset_inputs_covmat_t0_considered+ tmp.reindex(index=bb, columns=bb, level=0).values)
```
Comment (Zaharid) [inline validphys2/src/validphys/covmats.py:546]: Could we have the complicated reindexing as a separate function somewhere?

Comment (andreab1997) [reply validphys2/src/validphys/covmats.py:546]: Yes, I am doing that

File: validphys2/src/validphys/covmats.py [hunk 2]
```diff
@@ -690,12 +702,18 @@ def _covmat_t0_considered(covmat_t0_considered):
...
     covariance matrix accross dataset_inputs, having considered both ``use_t0``
     and ``use_pdferr``
     """
-    return dataset_inputs_covmat_t0_considered
+    if fitthcovmat is not None:
+        theory_covmat = fitthcovmat.load()
+        #change ordering according to exp_covmat (so according to runcard order)
+        tmp = theory_covmat.droplevel(0, axis=0).droplevel(0, axis=1)
+        bb = [str(i) for i in data]
+        return dataset_inputs_covmat_t0_considered + tmp.reindex(index=bb, columns=bb, level=0).values
```
Comment (Zaharid) [inline validphys2/src/validphys/covmats.py:715]: Could we use the new function in here as well? Also I have the impression one could encapsulate everything after `if fitthcovmat`

**Pull request merged**